p8e-api/src/main/kotlin/io/provenance/engine/event/IndexHandler.kt

```diff
@@ -114,6 +123,33 @@ class IndexHandler(
                                     envelope.data.result.contract.recitalsList.map { it.signer.encryptionPublicKey.toPublicKey().toSha512Hex() }
                                 ), RequestOptions.DEFAULT
                             )
+                            // If the env is the invoker, create the data access message and put into a job.
+                            if(envelope.isInvoker == true && envelope.data.input.affiliateSharesList.isNotEmpty()) {
+                                val envelopeDataAccess = envelope.data.input.affiliateSharesList.map {
+                                    affiliateService.getAddress(
+                                        it.toPublicKey(), chaincodeProperties.mainNet
```

Comment:
minor nit-picking after working with a bunch of keys, when following code you can get lost in the `it` when there is a bunch of nested `it`s ... so you can do `{ affiliateSharePublicKey -> affilliateSharePublicKey.toPublicKey() .. ` same thing for the ones below.